python/lsst/daf/butler/core/fileTemplates.py

```diff
@@ -552,9 +558,11 @@ def format(self, ref: DatasetRef) -> str:
                 "Component '{}' specified but template {} did not use it".format(component, self.template)
             )
 
-        # Complain if there's no run
-        if not usedRun:
-            raise KeyError("Template does not include 'run'.")
+        # Complain if there's no run or id
+        if not usedRun and not usedId:
+            missing = ("run" if not usedRun else None, "id" if not usedId else None)
+            text = " or ".join(f"'{m}'" for m in missing if m is not None)
+            raise KeyError(f"Template does not include {text}.")
```

Comment:
Is there a similar check for the dataset type name that could also be relaxed?

Reply:
It doesn't look like we require the dataset type to be in the template. "run" was the only thing we required from what I could see (and I was able to use a file template of just `{id}`.

Reply:
The fact that this line has no coverage implies that we catch the no-run logic higher up and none of this is needed any more.

Reply:
Sounds like we should just remove it, then, and update the docs to say it's expected?

Reply:
I think that this block of code was an early attempt at template validation but at some point we moved the validation logic earlier and never noticed that this block was irrelevant. I'll remove it.